File: frontend/lib/api/stats.ts
import { apiGet } from "@/lib/api/client";
import type { CategoryStat, MonthlyStat } from "@/types/stats";

export function getMonthlyStats(year?: number): Promise<MonthlyStat[]> {
  const qs = year ? `?year=${year}` : "";
  return apiGet<MonthlyStat[]>(`/stats/monthly${qs}`);
}

export function getCategoryStats(year?: number, month?: number): Promise<CategoryStat[]> {
  const query = new URLSearchParams();
  if (year !== undefined) query.set("year", String(year));
  if (month !== undefined) query.set("month", String(month));
  const qs = query.toString();
  return apiGet<CategoryStat[]>(`/stats/by-category${qs ? `?${qs}` : ""}`);
}

File: backend/app/api/routers/stats.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_db
from app.schemas.stats import CategoryStat, MonthlyStat
from app.services import stats_service

router = APIRouter(prefix="/stats", tags=["stats"])


@router.get("/monthly", response_model=list[MonthlyStat])
def monthly_stats(year: int | None = None, db: Session = Depends(get_db)) -> list[MonthlyStat]:
    return stats_service.monthly_totals(db, year=year)


@router.get("/by-category", response_model=list[CategoryStat])
def category_stats(
    year: int | None = None, month: int | None = None, db: Session = Depends(get_db)
) -> list[CategoryStat]:
    return stats_service.category_totals(db, year=year, month=month)
